contracts/programs/keystone-forwarder/src/lib.rs

```diff
@@ -0,0 +1,538 @@
+use anchor_lang::prelude::*;
+use anchor_lang::solana_program::{hash, keccak, secp256k1_recover::*};
+
+mod error;
+pub use error::*;
+
+declare_id!("whV7Q5pi17hPPyaPksToDw1nMx6Lh8qmNWKFaLRQ4wz");
+
+pub const STATE_VERSION: u8 = 1;
+
+pub const ANCHOR_DISCRIMINATOR: usize = 8;
+
+pub const REPORT_CONTEXT_LEN: usize = 96;
+
+pub const MAX_REPORT_SIGNERS: usize = 17;
+
+pub const SIGNATURE_LEN: usize = 65;
+
+pub const FORWARDER_METADATA_LENGTH: usize = 45;
+pub const METADATA_LENGTH: usize = 109;
+#[program]
+pub mod keystone_forwarder {
+    use anchor_lang::{
+        solana_program::{instruction::Instruction, program::invoke_signed, system_instruction},
+        Discriminator,
+    };
+
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        // store forwarder PDA bump for later
+        let (_authority_pubkey, authority_nonce) = Pubkey::find_program_address(
+            &[b"forwarder", ctx.accounts.state.key().as_ref()],
+            &crate::ID,
+        );
```

Comment:
nit: does this work? no need to name it

```suggestion
        let (_, authority_nonce) = Pubkey::find_program_address(
            &[b"forwarder", ctx.accounts.state.key().as_ref()],
            &crate::ID,
        );
```